mac.go

```diff
@@ -41,6 +43,9 @@ func doMac(macData *macData, message, password []byte) ([]byte, error) {
 	case macData.Mac.Algorithm.Algorithm.Equal(oidSHA256):
 		hFn = sha256.New
 		key = pbkdf(sha256Sum, 32, 64, macData.MacSalt, password, macData.Iterations, 3, 32)
+	case macData.Mac.Algorithm.Algorithm.Equal(oidSHA512):
+		hFn = sha512.New
+		key = pbkdf(sha512Sum, 32, 64, macData.MacSalt, password, macData.Iterations, 3, 32)
```

Comment:
The parameters need to be changed for use with SHA-512. I'm going to close this PR in favor of the `sha512` branch which has the correct parameters.